client/src/app/common/CustomRules/rules-utils.tsx

```diff
@@ -137,8 +144,10 @@ export const getLabels = (labels: string[]) =>
     { sourceLabel: "", targetLabel: "", otherLabels: [], allLabels: [] }
   );
 
-export const getRulesetTargetList = (ruleset: Ruleset) => {
-  return ruleset.rules.reduce((acc: string[], rule) => {
-    return [...acc, rule?.metadata?.target || ""];
-  }, []);
+export const getAllLabelsFromTarget = (target: Target) => {
+  return (
+    target?.labels?.reduce((acc: TargetLabel[], label) => {
+      return [...acc, label || ""];
+    }, []) || []
```

Comment:
This doesn't seem right.  `label` would be a `TagetLabel` so you're iterating to rebuild the source array?

Wouldn't it be that same to just do:
```js
    target?.labels ?? []
```

Further, it appears that this function is only used in `set-options.tsx` and only the `.label` field of the `TargetLabel` seems to be used.  Does the full object need to be there?  If it _is_ sufficient to just use `TargetLabel.label`, the code in both locations could be reduced, and this function could just be a simple pluck (and fancy with destructuring):
```js
    target?.labels?.map(({ label }) => label) ?? []
```

Reply:
Good catch. This function is a leftover from the previous implementation. Will update.